hooks/StrictNumericCastAnalyzer.php

```diff
@@ -39,9 +39,17 @@ public static function afterExpressionAnalysis(
                 continue;
             } elseif (
                 $previous_type instanceof TLiteralString &&
-                preg_match('#\d#', $previous_type->value[0] ?? '') // will probably have to inverse the check to forbid chars instead
+                // ' +5 foo' can be casted to int 5
+                // ' -.5 foo' can be casted to float -0.5
+                preg_match('#^\s*[+-]?\.?\d#', $previous_type->value)
```

Comment:
Do we really want to match strings with whitespace in front of the numeric?

Also, shouldn't we stop the regex at then end of the numeric to prevent ".0fefzefzefz" ?

Could you change your regex to allow "1.5" btw?

Reply:
> For me, the underlying reasoning of this error is to prevent casting of things that will result in 0

The result is not important, the reason for this error is to prevent casts that will result in a bug. I want to catch "zzz" to 0 the same way as I want to catch "555-1234" into "555" because you accidentaly casted a phone number into an int.

I feel like if you want to parse "10.5€" into a float, I feel like you should do it properly and prevent things like "10$" or "10.5€ ?? I'm not paying that much!" to be parsed as normal inputs

> This is already allowed with the regex used, isn't it?

You don't allow for any digit before the dot so, no.

Reply:
Actually, I'm not sure why we're bothering, we should just check is_numeric on the literal string. That way it will be the same standard as when we check for TNumericString in the previous condition